### PR TITLE
Replace usage of try! with ?

### LINKED ISSUE #330: Untry
Run https://github.com/japaric/untry on every `.rs` file in this repository to replace usage of the `try!(…)` macro with the `…?` operator.

### PR DESCRIPTION
**Pull request by SamWhited**
Replaces all uses of `try!` with the `?` operator.

Fixes #330

---
This change is [Reviewable](https://reviewable.io/reviews/servo/rust-url/331)

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -797,13 +797,13 @@ impl Url {
...
+            host: self.host()
                            .ok_or(())
-                           .or_else(|()| io_error("URL has no host"))),
-            port: try!(self.port_or_known_default()
+                           .or_else(|()| io_error("URL has no host"))?,
+            port: self.port_or_known_default()
                            .ok_or(())
                            .or_else(|()| f(self))
-                           .or_else(|()| io_error("URL has no port number")))
+                           .or_else(|()| io_error("URL has no port number"))?
```
Comment (SimonSapin) [inline src/lib.rs]: Please re-align these lines. (`untry` should have shown a warning about multi-line replacement.)

Comment (SamWhited) [inline src/lib.rs]: Fixed; is there a specific set of rustfmt settings that should be used? I tried to use it originally but it realigned a ton of other stuff so I left any sort of alignment issues alone. Thanks!

Comment (SimonSapin) [inline src/lib.rs]: rustfmt does a lot more things than just replacing `try!`. If we’re gonna go down that road I’d prefer to go all the way and enforce rustfmt-like style in CI. But that seems more disruptive than I wanted to get into right now, so I recommended untry instead.

**Comment by SimonSapin** [comment]
Great, thanks!

@bors-servo r+

File: src/lib.rs (hunk 1 above)
Comment (SamWhited) [inline src/lib.rs]: Ah, I see; untry gave me some warning about beign deprecated in favor of rustfmt; anyways, doesn't matter, fixed now. Thanks!

**Pull request merged**